spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -644,28 +646,43 @@ def add_electrical_series(recording: se.RecordingExtractor, nwbfile=None, metada
                 scalar_conversion = 1.
                 channel_conversion = gains * 1e-6
 
-            def data_generator(recording, channels_ids):
-                #  generates data chunks for iterator
-                for id in channels_ids:
-                    data = recording.get_traces(channel_ids=[id]).flatten()
-                    yield data
-
-            data = data_generator(recording=recording, channels_ids=channel_ids)
-            ephys_data = DataChunkIterator(data=data, iter_axis=1)
-
-            # To get traces in Volts = data*channel_conversion*conversion
-            ephys_ts = ElectricalSeries(
-                name=es_name,
-                data=ephys_data,
-                electrodes=electrode_table_region,
-                starting_time=recording.frame_to_time(0),
-                rate=recording.get_sampling_frequency(),
-                conversion=scalar_conversion,
-                channel_conversion=channel_conversion,
-                comments='Generated from SpikeInterface::NwbRecordingExtractor',
-                description=metadata['Ecephys']['ElectricalSeries'].get('description', defaults['description'])
+            if isinstance(recording.get_traces(), np.memmap):
+                ephys_data = H5DataIO(
+                    DataChunkIterator(
+                        data=np.memmap.transpose(recording.get_traces()),  # nwb standard is time as zero axis
```

Comment:
I think that `recording.get_traces().T` will automatically use `np.memmap.transpose` as `recording.get_traces()` is now a memmap object

Reply:
I suspect it would as well, but will triple check. I was basing this syntax off of what's in the read_binary helper function; would you like me to change it there as well?

Reply:
`recording.get_traces().transpose` should work too

Reply:
OK; tested it and both the .transpose and .T worked.